spec/features/ips/filter_ip_search_by_location_spec.rb

```diff
@@ -0,0 +1,23 @@
+describe 'filtering IP addresses by dynamic search' do
+  let!(:organisation) { create(:organisation) }
+  let(:user) { create(:user, organisation: organisation) }
+  let!(:location_one) { create(:location, organisation: organisation, address: 'Apple Street') }
+  let!(:location_two) { create(:location, organisation: organisation, address: 'Banana Road') }
+  let!(:location_three) { create(:location, organisation: organisation, address: 'Citrus Lane') }
+
+  before do
+    sign_in_user user
+    visit ips_path
+  end
+
+  context 'when user types in the search field' do
+    it 'displays the location that closely matches the text field input' do
+      fill_in 'filterInput', with: 'B'
+      # page.execute_script(filterLocations())
```

Comment:
Pseudocode for executing the javascript function.

Reply:
Could you not type in the search and click buttons through the UI to trigger this?

Reply:
The filter doesn't have a button that triggers the search. 
You type in the field and it automatically locates a match.